### PR TITLE
Oozelings - No Augments, Constantly Clean, Fleshmend Fix, Medibox Fix, Changeling Panacea Fix, Food Buff, Spellcheck fix, Rust Heretic fix

### PR DESCRIPTION
**Pull request by ro5490**
## About The Pull Request

Oozelings now are unable to get augmented, or get prosthetics(except from the roundstart trait)
Oozelings are now constantly clean, as they cant wash due to water melting them away.
Fixed changeling fleshmend hurting Oozlings
Fixed chaingeling panacea(Total Purge) from hurting Oozelings
Fixed Medibots doing tox damage instead of healing it on Oozelings
Fixed Rust Heretic regen damaging Oozelings.
Found a spelling issue in Hyperperspiration virus description while looking for code stuff, fixed that
Oozelings now actually gain a decent amount of blood back when they eat and get over-full

## Why It's Good For The Game
Bugfixes, Spellcheck and balance changes are always worth the effort.

## Changelog
:cl:
add: Just a reminder, water in any form is LETHAL to Oozelings, Be careful what you eat.
add: Oozelings are now always clean!
tweak: Adjusted the blood volume gain from eating well, and becoming full (plus a fullness reduce after a certain threshold is met).
balance: Oozelings can no longer be augmented.
fix: Fixed Fleshmend damaging Oozelings.
fix: Fixed Rust Heretic regen damaging Oozelings.
fix: Fixed medibots damaging Oozelings when trying to heal toxin damage
spellcheck: Hyperperspiration's description now reads "extinguishing small fires" instead of "extnguishing small fires"

/:cl:

### CODE REVIEW

**Comment by utilbird** [comment]
Shouldn't oozelings get dirty even quicker than other races, rather than always being clean. Considering they're made out of slime

**Comment by ro5490** [comment]
> Shouldn't oozelings get dirty even quicker than other races, rather than always being clean. Considering they're made out of slime

The main issue with this is that almost all cleaning methods require water, and well... you know how water works with these guys lmao

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm [hunk 1]
```diff
@@ -49,8 +49,12 @@
 		H.blood_volume += 5
 		H.adjustBruteLoss(5)
 		to_chat(H, "<span class='danger'>You feel empty!</span>")
+	if(H.nutrition >= NUTRITION_LEVEL_ALMOST_FULL)
+		H.adjust_nutrition(-5)
+		if(H.blood_volume <= 672)
+			H.blood_volume += 10
 	if(H.nutrition >= NUTRITION_LEVEL_WELL_FED && H.blood_volume <= 672)
-		H.blood_volume += 1
+		H.blood_volume += 8
```
Comment (utilbird) [inline code/modules/mob/living/carbon/human/species_types/oozelings.dm]: Being almost full will run both of these (no else if), for a total of 18 blood per tick. Not sure if that's intentional or not

File: code/modules/surgery/limb_augmentation.dm [hunk 2]
```diff
@@ -43,6 +43,13 @@
 	possible_locs = list(BODY_ZONE_R_ARM,BODY_ZONE_L_ARM,BODY_ZONE_R_LEG,BODY_ZONE_L_LEG,BODY_ZONE_CHEST,BODY_ZONE_HEAD)
 	requires_real_bodypart = TRUE
 
+
+
+/datum/surgery/augmentation/can_start(mob/user, mob/living/carbon/target)
+	if(!isoozeling(target))
+		return TRUE
+	return FALSE
```
Comment (utilbird) [inline code/modules/surgery/limb_augmentation.dm]:
```suggestion
	return !isoozeling(target)
```

File: code/datums/status_effects/buffs.dm [hunk 3]
```diff
@@ -334,8 +334,11 @@
 	owner.adjustBruteLoss(-2.5, FALSE)
 	owner.adjustFireLoss(-2.5, FALSE)
 	owner.adjustOxyLoss(-2.5)
-	owner.adjustToxLoss(-2.5, FALSE)
 	owner.adjustCloneLoss(-2.5)
+	if(isoozeling(owner))
```
Comment (Archanial) [inline code/datums/status_effects/buffs.dm]: This all should be done with special racial trait, not just by checking for race

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm (hunk 1 above)
Comment (ro5490) [reply code/modules/mob/living/carbon/human/species_types/oozelings.dm]: This was intentional, if it needs adjusting however I'll tweak it a bit

**Comment by utilbird** [comment]
> The main issue with this is that almost all cleaning methods require water, and well... you know how water works with these guys lmao

I suppoose, oozelings don't have that many downsides relative to their upsides. This could be treated as a mechanical downside rather than an oversight, i.e plasmaman cloning procedure. Or you could just give them their own unique, more difficult way of cleaning (space cleaner, etc)

That's my take at least xd

**Review by PowerfulBacon — DISMISSED** [review]

**Comment by ivanmixo** [comment]
They also take damage from rust in the heretic rust path I'm pretty sure. Same goes for regular slime people.

**Comment by ro5490** [comment]
> They also take damage from rust in the heretic rust path I'm pretty sure. Same goes for regular slime people.

As in, as a heretic on rust path, they take damage from the `rust_regen`?

**Review by utilbird — DISMISSED** [review]

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm (hunk 1 above)
Comment (utilbird) [reply code/modules/mob/living/carbon/human/species_types/oozelings.dm]: Well, it's pretty fast, especially for a race that can utilize high blood levels

File: code/datums/status_effects/buffs.dm (hunk 3 above)
Comment (ro5490) [reply code/datums/status_effects/buffs.dm]: its meant to be handled by the toxinlover trait, but for whatever reason, changing the `FALSE` to `TRUE` which is meant to do the thing, doesnt actually do the thing

Comment (park66665) [reply code/datums/status_effects/buffs.dm]: I have no idea why you are doing this. Oozeling's `TRAIT_TOXINLOVER` trait works on `adjustToxLoss` just as intended.

File: code/modules/surgery/limb_augmentation.dm [hunk 4]
```diff
@@ -43,6 +43,11 @@
 	possible_locs = list(BODY_ZONE_R_ARM,BODY_ZONE_L_ARM,BODY_ZONE_R_LEG,BODY_ZONE_L_LEG,BODY_ZONE_CHEST,BODY_ZONE_HEAD)
 	requires_real_bodypart = TRUE
 
+
+
+/datum/surgery/augmentation/can_start(mob/user, mob/living/carbon/target)
+	return !isoozeling(target)
```
Comment (park66665) [inline code/modules/surgery/limb_augmentation.dm]:
```suggestion
/datum/surgery/augmentation/can_start(mob/user, mob/living/carbon/target)
	return ..() && !isoozeling(target)
```

Comment (utilbird) [reply code/modules/surgery/limb_augmentation.dm]: @park66665 Would parent call be needed for this? most non-advanced surgeries don't call

**Comment by BitBite112** [comment]
What if they're sick with a disease and the cure is water?

File: code/modules/surgery/limb_augmentation.dm (hunk 4 above)
Comment (park66665) [reply code/modules/surgery/limb_augmentation.dm]: `/datum/surgery/proc/can_start` is actually a reasonably complex set of checks which has been in use for augmentation surgery. I wouldn't override that.

**Comment by Crossedfall** [comment]
> 
> 
> What if they're sick with a disease and the cure is water?

Vaccines

File: code/datums/status_effects/buffs.dm (hunk 3 above)
Comment (ro5490) [reply code/datums/status_effects/buffs.dm]: When I tested this for the medibot and fleshmend fix, it didn't seem to work at all. 
I'll retest it tonight anyway but yeah

Comment (ro5490) [reply code/datums/status_effects/buffs.dm]: image
Thats with this
image

File: code/modules/mob/living/carbon/human/species_types/oozelings.dm (hunk 1 above)
Comment (ro5490) [reply code/modules/mob/living/carbon/human/species_types/oozelings.dm]: adjusting it to
```
	if(H.nutrition >= NUTRITION_LEVEL_WELL_FED && H.blood_volume <= 672)
		if(H.nutrition >= NUTRITION_LEVEL_ALMOST_FULL)
			H.adjust_nutrition(-5)
			H.blood_volume += 10
		else
			H.blood_volume += 8
			```

**Comment by ro5490** [comment]
I think that's most of the bugs fixed,  I don't plan to add anything else to this except any code improvement suggestions

**Review by utilbird — APPROVED** [review]

**Pull request merged**